core/src/main/java/com/novoda/noplayer/SubtitleView.java

```diff
@@ -0,0 +1,84 @@
+package com.novoda.noplayer;
+
+import android.content.Context;
+import android.graphics.Canvas;
+import android.util.AttributeSet;
+import android.view.View;
+
+import com.novoda.noplayer.model.NoPlayerCue;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public final class SubtitleView extends View {
+
+    private static final float DEFAULT_TEXT_SIZE_FRACTION = 0.0533f;
+    private static final float DEFAULT_BOTTOM_PADDING_FRACTION = 0.08f;
+
+    private static final boolean APPLY_EMBEDDED_STYLES = true;
+    private static final boolean APPLY_EMBEDDED_FONT_STYLES = true;
+
+    private static final int ZERO_PIXELS = 0;
+    private static final int NO_CUES = 0;
+
+    private final List<SubtitlePainter> painters;
+
+    private List<NoPlayerCue> cues;
+
+    public SubtitleView(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        painters = new ArrayList<>();
+    }
+
+    public void setCues(List<NoPlayerCue> cues) {
+        if (this.cues == cues) {
```

Comment:
in the case of long piece of content a quick hash check will be a lot faster than chaining `equals()` to all the children